R/location_packit.R

```diff
@@ -0,0 +1,75 @@
+github_oauth_client <- function() {
+  # Surprisingly, we don't actually need the Client ID here to match the one
+  # used by Packit. It should be fine to hardcode a value regardless of which
+  # server we are talking to.
+  httr2::oauth_client(
+    id = "Ov23liUrbkR0qUtAO1zu",
+    token_url = "https://github.com/login/oauth/access_token",
+    name = "orderly2"
+  )
+}
+
+do_oauth_device_flow <- function(base_url) {
```

Comment:
It's unfortunately a httr2 thing, but I was confused the first time seeing the prompt:

```
ℹ Logging in to https://packit.dide.ic.ac.uk/malariaverse-sitefiles/
→ Copy 18C0-2805 and paste when requested by the browser
Press <enter> to proceed:
```

At this point I thought that I was waiting for a browser window to open, then I tried clicking the link. Finally I pressed enter and it launched the browser.  It might be worth here (or around l39) posting a small hint about how the login process will go?

Reply:
Yeah this is quite annoying. It also doesn't work if R can't open a browser, eg. over SSH.
I'll fix something.

Reply:
Alright so if I override interactive temporarily, the behaviour is a lot less confusing. I've changed it, let me know if this helps.